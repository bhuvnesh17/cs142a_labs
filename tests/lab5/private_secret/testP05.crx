// test all the pre-defined functions

func main() : void {
  ::printInt();
  ::printFloat();
  ::printBool();

  var x : float;
  let x = ::readInt(7);

  var y : int;
  let y = ::readFloat(7.8);
}
