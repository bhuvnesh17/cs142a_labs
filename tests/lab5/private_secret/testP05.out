Error type-checking file.
TypeError(4,3)[Cannot call func(TypeList(int)):void using TypeList().]
TypeError(5,3)[Cannot call func(TypeList(float)):void using TypeList().]
TypeError(6,3)[Cannot call func(TypeList(bool)):void using TypeList().]
TypeError(9,11)[Cannot call func(TypeList()):int using TypeList(int).]
TypeError(9,3)[Cannot assign ErrorType(Cannot call func(TypeList()):int using TypeList(int).) to Address(float).]
TypeError(12,11)[Cannot call func(TypeList()):float using TypeList(float).]
TypeError(12,3)[Cannot assign ErrorType(Cannot call func(TypeList()):float using TypeList(float).) to Address(int).]

